File: sys/src/inference/symbol.cc
#include "symbol.h"

namespace gtsam {

std::unique_ptr<Symbol> default_symbol() { return std::make_unique<Symbol>(); }

std::unique_ptr<Symbol> new_symbol(unsigned char character, uint64_t index) {
    return std::make_unique<Symbol>(character, index);
}

} // namespace gtsam

File: src/inference/symbol.rs
use cxx::UniquePtr;
use gtsam_sys::Key;

pub struct Symbol {
    inner: UniquePtr<::sys::Symbol>,
}

impl Default for Symbol {
    fn default() -> Self {
        Self {
            inner: ::sys::default_symbol(),
        }
    }
}

impl Symbol {
    pub fn new(character: u8, index: u64) -> Self {
        Self {
            inner: ::sys::new_symbol(character, index),
        }
    }

    pub(crate) fn key(&self) -> Key {
        self.inner.key()
    }
}
